libultraship/libultraship/ImGuiImpl.cpp

```diff
@@ -592,31 +595,43 @@ namespace SohImGui {
         else
             ImGui::Text(text, static_cast<int>(100 * val));
 
+        InsertPadding();
+
         if(PlusMinusButton) {
             std::string MinusBTNName = " - ##";
             MinusBTNName += cvarName;
             if (ImGui::Button(MinusBTNName.c_str())) {
-                val -= 0.1f;
+                if (!isPercentage)
+                    val -= 0.1f;
+                else
+                    val -= 0.01f;
                 CVar_SetFloat(cvarName, val);
                 needs_save = true;
             }
             ImGui::SameLine();
             ImGui::SetCursorPosX(ImGui::GetCursorPosX() - 7.0f);
         }
-
+        if (PlusMinusButton) {
+            ImGui::PushItemWidth(ImGui::GetWindowSize().x - 79.0f);
+        }
         if (ImGui::SliderFloat(id, &val, min, max, format))
         {
             CVar_SetFloat(cvarName, val);
             needs_save = true;
         }
-
+        if (PlusMinusButton) {
+            ImGui::PopItemWidth();
+        }
         if(PlusMinusButton) {
             std::string PlusBTNName = " + ##";
             PlusBTNName += cvarName;
             ImGui::SameLine();
             ImGui::SetCursorPosX(ImGui::GetCursorPosX() - 7.0f);
             if (ImGui::Button(PlusBTNName.c_str())) {
-                val += 0.1f;
+                if (!isPercentage)
+                    val += 0.1f;
+                else
+                    val += 0.01f;
```

Comment:
didn't see this mentioned in the pr description, is this a bugfix?

Reply:
Kind of. When I added the plus/minus buttons to the internal resolution slider, it would jump up by 10% at a time instead of 1%. It's just that there wasn't any float sliders before that used percentages.

Reply:
Cool!